### ISSUE TITLE
[Tiled] Layers rendering incorrectly when not square

### ISSUE THREAD

**Issue by TheUltimateHose** [craftworkgames/MonoGame.Extended#235]
I was recently messing around with Tiled map rendering and came across the issue that when I tried to render a layer that was 16x19 tiles large, only the first 16x16 tiles would actually be rendered. The issue was solved by changing the map size to 19x19, but that is not really a great solution.

The code I used to render the layer looks like this:

```
_spritebatch.Start(transformMatrix: camera.GetViewMatrix(), samplerState: SamplerState.PointClamp);
_layer.Draw(_spritebatch);
_spritebatch.End();
```

And the followong code produces the same result:

```
_spritebatch.Start(transformMatrix: camera.GetViewMatrix(), samplerState: SamplerState.PointClamp);
_spritebatch.Draw(_layer);
_spritebatch.End();
```

As mentioned above, the three rows on the bottom would not render, but I didn't test that with different render orders (mine was right-down).

**EDIT: I just found that this bug was already reported (issue was closed)**

**Comment by craftworkgames** [comment]
We've been working on the Tiled renderer a lot lately. I'm not sure if the bug exists in the new renderer or not but just in case I'll leave this issue open.

Would you be able to provide a TMX file to reproduce the issue?

**Comment by TheUltimateHose** [comment]
[Here](https://gist.github.com/TheUltimateHose/b86d8d1f8841d5c58a954c1e15dd6c24)'s a gist with the TMX file, the texture can be found [here](http://kenney.nl/assets/roguelike-modern-city).

**Comment by TheUltimateHose** [comment]
I just downloaded and updated to the latest available package from your build server (0.5.415-beta) and tested it myself and the above code worked just fine, rendering the entire map even if it's not square.

**Comment by HyperionMT** [comment]
Can this be closed?  I was going to fix it as part of #268 but couldn't reproduce it.  I've also added a new level04.tmx file (non-square map) to the TiledMaps demo as part of that issue so we can make sure it doesn't break again.

**Comment by craftworkgames** [comment]
> Can this be closed? I was going to fix it as part of #268 but couldn't reproduce it. I've also added a new level04.tmx file (non-square map) to the TiledMaps demo as part of that issue so we can make sure it doesn't break again.

Yeah, let's close it. If you can't reproduce it's most likely fixed. Besides, it's always difficult to fix things if you can't prove they are broken ;)
